pontoon/teams/views.py

```diff
@@ -379,6 +384,90 @@ def ajax_translation_memory_delete(request, locale):
     return HttpResponse("ok")
 
 
+@require_AJAX
+@require_POST
+@permission_required_or_403("base.can_translate_locale", (Locale, "code", "locale"))
+@transaction.atomic
+def ajax_translation_memory_upload(request, locale):
+    """Upload Translation Memory entries from a .TMX file."""
+    try:
+        file = request.FILES["tmx_file"]
+    except MultiValueDictKeyError:
+        return JsonResponse(
+            {"status": False, "message": "No file uploaded."},
+            status=400,
+        )
+
+    if not file.name.endswith(".tmx"):
+        return JsonResponse(
+            {
+                "status": False,
+                "message": "Invalid file format. Only .TMX files are supported.",
+            },
+            status=400,
+        )
+
+    locale = get_object_or_404(Locale, code=locale)
+    code = locale.code
+
+    # Parse the TMX file
+    try:
+        tree = ET.parse(file)
+        root = tree.getroot()
+    except ET.ParseError as e:
+        return JsonResponse(
+            {"status": False, "message": f"Invalid XML file: {e}"}, status=400
+        )
+
+    # Extract TM entries
+    file_entries = []
+    ns = {"xml": "http://www.w3.org/XML/1998/namespace"}
+    for tu in root.findall(".//tu"):
+        try:
+            source = tu.find("./tuv[@xml:lang='en-US']/seg", namespaces=ns).text.strip()
```

Comment:
We should read the source locale code from the `<header srclang>` attribute and use it here. We should support `en`, `en-*` and `en_*` in any capitalization.

We should also support the `lang` attribute (in addition to `xml:lang`).

Reply:
I'm confused by this comment: what does "we should" mean? This patch needs to be updated, follow-up?

> We should read the source locale code from the <header srclang> attribute and use it here.

The translation unit can also have a `srclang` attribute.

Reply:
> I'm confused by this comment: what does "we should" mean? This patch needs to be updated, follow-up?

I'll update this PR.

> The translation unit can also have a `srclang` attribute.

I would ignore that, at least for the first iteration.

Reply:
Thanks, I'll build upon that.